File: 3345-smallest-divisible-digit-product-i/3345-smallest-divisible-digit-product-i.py
class Solution:
    def smallestNumber(self, n: int, t: int) -> int:
        def check(n: int, t: int):
            prod = 1
            for c in str(n):
                prod *= ord(c) - ord("0")
            return prod % t == 0

        while not check(n, t):
            n += 1
        return n

File: 3345-smallest-divisible-digit-product-i/3345-smallest-divisible-digit-product-i.cpp
class Solution {
public:
    int smallestNumber(int n, int t) {
        
        auto check = [&](int n)
        {
            string s = to_string(n);
            int x = 1;
            for (auto c : s)
                x *= c - '0';
            return x % t == 0;
        };
        
        while (1)
        {
            if (check(n)) break;
            n += 1;
        }
        
        return n;
    }
};